src/mapper/pkg/resolvers/helpers.go

```diff
@@ -99,7 +101,17 @@ func (i *IntentsHolder) AddIntent(srcService model.OtterizeServiceIdentity, dstS
 		intents = goset.NewSet[model.OtterizeServiceIdentity]()
 		i.store[srcService] = intents
 	}
-	intents.Add(model.OtterizeServiceIdentity{Name: dstService.Name, Namespace: namespace})
+	intent := model.OtterizeServiceIdentity{Name: dstService.Name, Namespace: namespace}
+	if !intents.Contains(intent) {
+		i.lastIntentsUpdate = time.Now()
+	}
+	intents.Add(intent)
```

Comment:
Move the `Add` into the if, because there's no reason to call it if it is already contained :)